mutect2Consensus.wdl

```diff
@@ -446,8 +489,8 @@ runtime {
 }
 
 output {
-  File annotatedCombinedVcf = "~{basename}.merged.vcf.gz"
-  File annotatedCombinedIndex = "~{basename}.merged.vcf.gz.tbi"
+  File annotatedCombinedVcf = "~{outputPrefix}.merged.vcf.gz"
```

Comment:
This workflow has tumorInputGroup and normalInputGroup,  I only supplied the tumor file as for the outputFileNamePrefix for the workflow, this is same strategy mutect2 (including matched version) is using now. But mutect2Consensus runs mutect2 tumor only mode for both tumor and normal inputGroup, so one run is "normal only" but mutect2 will still have "tumor_only" in the output file suffix (so that's why used getFileName task to remove those suffix). 
If I don't supply two outputFileNamePrefixes for this workflow, one group of output files needs get the output prefix from base name. For example normal output group use base name generated prefix.
Currently I call mutect2, combinvariant, annotaion, vep using base name generated prefix, only filterMaf used input outputFileNamePrefix from workflow input because it doesn't need run on normal version.
The mutec2 and vep pulled-in version does not have an input "outputFileNamePrefix". I see the new version of mutect2 has this as input parameter, but if update this subworkflow version than tedious parameter changes and regression file parameters changes needed. Vep still don't have ouptFileNamePrefix input. 

One more reason I need base name generated output prefix: I used scatter to run tumor/nomal input group, so the colleted result is an array, no direct indication of which one is from tumor or normal input, so need find it from output base name.

Reply:
Why not to append the basename-derived prefix to user-supplied **outputFileNamePrefixes**? Consider this scenario: I have an **outputFileNamePrefixes** which points to a custom directory: /.mounts/labs/gsiprojects/my_project/MYTEST_. If you using basenames as prefix the results will go elsewhere (research output directory, for example). Appending basename: /.mounts/labs/gsiprojects/my_project/MYTEST_**PCSI_0001_WG_R**_normal_only (for normal-only) or /.mounts/labs/gsiprojects/my_project/MYTEST_**PCSI_0001_WG_T** (for matched) or /.mounts/labs/gsiprojects/my_project/MYTEST**_PCSI_0001_WG_T**_tumor_only. And this should be used for the final results (Files in the outputs section of the workflow). All intermediate files may use whatever prefix is practical. If the current versions of pulled workflows do not use outputFileNamePrefixes it does not mean it's right

Reply:
My intention for using base name for the file name prefix is because at some section the code has no way to determine which should be the prefix, the tumor or the normal, and base name would not include file path. Using **outputFileNamePrefixes** need to determine tumor or normal in advance.
As for you concerned about final output file path, I don't know why file name could influence file path, I have sreached in FPR the see the output path for some samples, all ouput files are under same directory: "/.mounts/labs/prod/vidarr/output-research"

Reply:
If you mean your scatter block (line 89) where tumor and normal inputs are in the same array, then there is a solution to this - you may break that scatter in two, one for tumor and one for normal. They will still run in parallel. I think that my example above illustrates the issue with the current implementation: if I want to output my files into a directory **other than** "/.mounts/labs/prod/vidarr/output-research" it would be impossible with the current implementation. I don't like tedious things too, but it looks like updating pulled workflows is a right thing to do.

Reply:
if breaks  line 89 scatter (ig in inputGroups) in two, (which means not using scatter for tumor/normal inputs), then need redundantly double code lines, and with wdl's syntax restrcition on variable name, I need rename all the avaraibles to avoid repeats. Much effort to do ugly. 
I don't understand the issue of my current naming. actually this naming changing is in your previouly proved PR (which was not merged), I am just using a new PR to incoporate updates from Aqsa on separate branch.

Reply:
I dont see an issue in the way Gavin is using basename. Looks like this was implemented to be able to run mutect2 on normal and tumor, and have appropriate output names. 
we shouldn't use a file path in "outputFileNamePrefix", as it is a prefix to name the output. If the outputs need to be provisioned out elsewhere that should be done using cromwells "options.json" to specify the output directory. 
I do however have some concerns on the output names, I'll comment on the code to make it more clear but generally.

1. It's unclear where the filterredMaf is coming from, is this from matched, tumor only or normal mode? filterredMaf also has a typo

Reply:
1, the filteredMaf ( I'll correct the typo) file comes from filterMaf task which have both tumor and normal maf inputs.  So in terms of filterMaf task, it is matched, but all the previous steps including muetct2 and vep, they are tumor only (or normal only). There is another matchedFilterredMaf which comes from the filterMaf task with only mafFile input no nroamlMaf input, because the mafFile already combined tumor and normal inputs in previous mutect2 and vep tasks.
This is workflow includes multiple tasks and inputs, the naming become intriguing, the line of is which is tumor only, which is matched becomes blurring. In theory filteredMaf and matchedFilteredMaf should be identical, but also could be different given they run through different processes
2, GroupID missing because my manual input missed that, in olive it won't miss
3, tumor_only label can be removed if I add one task to specifically remove that, it comes from muetct2 output suffix

Reply:
An yes I agree we can remove outputFileNamePrefix.

Reply:
Having **outputFileNamePrefix** is a common best practice and I do not see why this workflow needs to be an exception. We expect certain behaviour from a workflow, and in this case we won't have the outputs provisioned consistently (outside of cromwell/shesmu environment). I think I explained it in great detail above.

Reply:
@pruzanov You already agreed outputFileNamePrefix does not contain file path, and basename by definition deos not comtain file path, so the two equally serves well as file prefix. This is true in wdl and olive, we never have the concern of file path carried over to the filnal output causes inconsisiantcy of output directory, so this is not a problem need to be fixed. Do you have other reason use outputFileNamePrefix is a better or best practice?

Reply:
I never said that **outputFileNamePrefix** should not contain a file path, the problem with this workflow is that it does not support the usage of **outputFileNamePrefix** consistently, as outlined in our Best Practices. This is something that has been established long time ago, and it needs to be fixed, that's all.

Reply:
The best practice is just a habit or to avoid some errors, concerns? since there is no problem caused here ( and I remove the outputFileNamePrefix, so no "inconsistant use" here), just follow the habit without more benefit, but very costly to change the wdl, I think we are better off not using it.